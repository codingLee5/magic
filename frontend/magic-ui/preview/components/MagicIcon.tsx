import React from "react"
import { Space } from "antd"
import MagicIcon from "../../components/MagicIcon"
import ComponentDemo from "./Container"
import { IconSearch, IconFolder, IconSettings, IconUser, IconHome } from "@tabler/icons-react"

const IconDemo: React.FC = () => {
	return (
		<div>
			<ComponentDemo
				title="基础图标"
				description="最基本的图标组件"
				code="<MagicIcon name='IconHome' />"
			>
				<Space>
					<MagicIcon component={IconHome} />
					<MagicIcon component={IconUser} />
					<MagicIcon component={IconSettings} />
					<MagicIcon component={IconSearch} />
				</Space>
			</ComponentDemo>

			<ComponentDemo
				title="不同尺寸"
				description="支持不同尺寸的图标"
				code="size: number | string"
			>
				<Space>
					<MagicIcon component={IconHome} size={16} />
					<MagicIcon component={IconHome} size={24} />
					<MagicIcon component={IconHome} size={32} />
					<MagicIcon component={IconHome} size={48} />
				</Space>
			</ComponentDemo>

			<ComponentDemo title="不同颜色" description="支持不同颜色的图标" code="color: string">
				<Space>
					<MagicIcon component={IconHome} color="#1890ff" />
					<MagicIcon component={IconHome} color="#52c41a" />
					<MagicIcon component={IconHome} color="#faad14" />
					<MagicIcon component={IconHome} color="#f5222d" />
				</Space>
			</ComponentDemo>

			<ComponentDemo
				title="描边宽度"
				description="自定义图标的描边宽度"
				code="stroke: number"
			>
				<Space>
					<MagicIcon component={IconHome} stroke={1} />
					<MagicIcon component={IconHome} stroke={2} />
					<MagicIcon component={IconHome} stroke={3} />
				</Space>
			</ComponentDemo>

			<ComponentDemo title="自定义组件" description="使用自定义的图标组件" code="component">
				<Space>
					<MagicIcon component={IconSearch} />
					<MagicIcon component={IconFolder} />
					<MagicIcon component={IconSettings} />
					<MagicIcon component={IconUser} />
				</Space>
			</ComponentDemo>
		</div>
	)
}

export default IconDemo
