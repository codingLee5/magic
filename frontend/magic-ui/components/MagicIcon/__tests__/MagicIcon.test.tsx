import { render } from "@testing-library/react"
import MagicIcon from "../index"
import MagicThemeProvider from "../../ThemeProvider"
import { describe, it, expect } from "vitest"

const renderWithTheme = (component: React.ReactElement) =>
	render(<MagicThemeProvider theme="light">{component}</MagicThemeProvider>)

describe("MagicIcon", () => {
	it("应该正常渲染", () => {
		renderWithTheme(<MagicIcon name="user" />)
		expect(true).toBe(true)
	})

	// 快照测试
	describe("快照测试", () => {
		it("基础图标快照", () => {
			const { asFragment } = renderWithTheme(<MagicIcon name="user" />)
			expect(asFragment()).toMatchSnapshot()
		})

		it("不同图标快照", () => {
			const { asFragment } = renderWithTheme(<MagicIcon name="home" />)
			expect(asFragment()).toMatchSnapshot()
		})

		it("带自定义样式图标快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicIcon name="user" style={{ color: "red", fontSize: "20px" }} />,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("带类名图标快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicIcon name="user" className="custom-icon" />,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("大尺寸图标快照", () => {
			const { asFragment } = renderWithTheme(<MagicIcon name="user" size="large" />)
			expect(asFragment()).toMatchSnapshot()
		})

		it("小尺寸图标快照", () => {
			const { asFragment } = renderWithTheme(<MagicIcon name="user" size="small" />)
			expect(asFragment()).toMatchSnapshot()
		})

		it("带点击事件图标快照", () => {
			const { asFragment } = renderWithTheme(<MagicIcon name="user" onClick={() => {}} />)
			expect(asFragment()).toMatchSnapshot()
		})
	})
})
