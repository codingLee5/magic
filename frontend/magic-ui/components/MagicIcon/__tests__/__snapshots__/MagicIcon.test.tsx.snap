// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MagicIcon > 快照测试 > 不同图标快照 1`] = `<DocumentFragment />`;

exports[`MagicIcon > 快照测试 > 基础图标快照 1`] = `<DocumentFragment />`;

exports[`MagicIcon > 快照测试 > 大尺寸图标快照 1`] = `<DocumentFragment />`;

exports[`MagicIcon > 快照测试 > 小尺寸图标快照 1`] = `<DocumentFragment />`;

exports[`MagicIcon > 快照测试 > 带点击事件图标快照 1`] = `<DocumentFragment />`;

exports[`MagicIcon > 快照测试 > 带类名图标快照 1`] = `<DocumentFragment />`;

exports[`MagicIcon > 快照测试 > 带自定义样式图标快照 1`] = `<DocumentFragment />`;
